tesseract_core/runtime/schema_types.py

```diff
@@ -356,6 +357,61 @@ def is_differentiable(obj: Any) -> bool:
     return False
 
 
+class FileReference:
```

Comment:
Just reading this before our meeting and leaving my thoughts here so they do not get lost (sorry for commenting on a draft). Maybe we can do this with a field validator?
```python
from typing import Annotated
from pydantic import AfterValidator, BaseModel, ValidationError

def is_relative(path: Path) -> Path:
        if path.is_relative_to(Path.cwd()):
            path = path.relative_to(Path.cwd())
        else:
            raise ValueError(
                f"FileReference path must be relative to the current working directory: {path}"
            )
    return path


class FileReference(BaseModel):
    path: Annotated[Path, AfterValidator(is_relative)]
```

Reply:
I played around with RootModel a bit. The key issue I run into is that for a schema like

```
class InputSchema(BaseModel):
    name: str = Field(description="Name of the person you want to greet.")
    input_file: FileReference = Field(description="A file that can be used as input.")
```
the server gets the validated model
```
name='Osborne' input_file=Apply_FileReference(root=PosixPath('test.txt'))
```
I don't see a straightforward way to get rid of the nested model and resolve straight to a path. In order to avoid nesting, my understanding is that we actually have to go with the custom type (which also has the upside that we can even pass plain strings and automatically convert them to Path objects).

Reply:
actually, what about implementing `__getattr__` like this
```python
class FileReference(RootModel):
    root: Annotated[Path, AfterValidator(is_relative)]

    def __getattr__(self, key):
        return getattr(self.root, key)
```
wouldn't that give us what we need? I think that should be fair game for pydantic models?